framework/src/media/InputHandler.cpp

```diff
@@ -57,11 +61,72 @@ bool InputHandler::doStandBy()
 
 	std::thread wk = std::thread([=]() {
 		medvdbg("InputHandler::doStandBy thread enter\n");
-		player_event_t event;
-		if (mInputDataSource->open()) {
-			event = PLAYER_EVENT_SOURCE_PREPARED;
+		player_event_t event = PLAYER_EVENT_SOURCE_OPEN_FAILED;
+		if (!mInputDataSource->open()) {
+			meddbg("Open data source failed!\n");
 		} else {
-			event = PLAYER_EVENT_SOURCE_OPEN_FAILED;
+			/*
+			 * In case of BufferInputDataSource, even if the open operation succeed,
+			 * audio informations (audioType/channels/sampleRate/pcmFormat) may be
+			 * unknown. It's necessary to preload and parse data to get the audio
+			 * informations if application didn't specify all of them explicitly.
+			 */
+			audio_type_t audioType = mInputDataSource->getAudioType();
+			unsigned int channels = mInputDataSource->getChannels();
+			unsigned int sampleRate = mInputDataSource->getSampleRate();
+			// Release present preload buffer if exist
+			if (mPreloadDataBuff) {
```

Comment:
Actually open means that openning source(connect network(httpDataSource), open file(fileDataSource), alloc buf(buf data source) where to read or write.
```
mPreloadDataBuff = malloc();
mInputDataSource->open();
while (readsize < threashold) {
     mInputDataSource->read();
     inputHandler->copy();
     audioType = utils::getAudioTypeFromStream(bufptr, (size_t)buflen);
     if (audioType == AUDIO_TYPE_UNKNOWN) {
           continue;
     }
     utils::buffer_header_parsing();
     mInputDataSource->setAudioType(audioType);
     ...
     if (channel == 0 || sampleRate ==0) { //check data type now.
            continue;
     }
```

Reply:
Changed to the logic as you said, please help to review again. Thanks.